### PR TITLE
added job history status view

### PR DESCRIPTION
**Pull request by adityathebe**
Example query res: 

```
select id, name, job_created_at, job_status FROM templates_with_status
+--------------------------------------+--------------+----------------------------+------------+
| id                                   | name         | job_created_at             | job_status |
|--------------------------------------+--------------+----------------------------+------------|
| 36953751-d008-456e-a8f8-119b870ab76f | aditya-thebe | 2023-03-14 13:44:22.007021 | FINISHED   |
| 018678f9-d892-989a-0822-fcf087028007 | petclinic    | 2023-03-14 13:44:22.019511 | FINISHED   |
+--------------------------------------+--------------+----------------------------+------------+
```

Closes https://github.com/flanksource/duty/issues/52

### CODE REVIEW

**Comment by moshloop** [comment]
@adityathebe I am expecting a view per resource - we are adding the status fields into the existing tables so we need the existing data.

- templates
- canaries
- scrapers
- teams

File: views/014_job_history_status.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
+CREATE
+OR REPLACE VIEW job_history_latest_status AS
+SELECT
+  jh.*
```
Comment (moshloop) [inline views/014_job_history_status.sql:4]: don't use abbreviations / acronyms

File: views/014_job_history_status.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+FROM
+  job_history jh
+  JOIN (
+    SELECT
+      resource_id,
+      MAX(created_at) AS max_created_at
```
Comment (moshloop) [inline views/014_job_history_status.sql:10]: We also need status, errors, duration, etc.

File: views/014_job_history_status.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+-- Template View
+CREATE
+OR REPLACE VIEW templates_with_status AS
+SELECT
+  t.*,
+  jhls.created_at job_created_at,
+  jhls.status job_status
+FROM
+  templates t
+  LEFT JOIN job_history_latest_status jhls ON t.id :: TEXT = jhls.resource_id;
```
Comment (moshloop) [inline views/014_job_history_status.sql:27]: I think we need to filter on resource type as well

**Review by moshloop — CHANGES_REQUESTED** [review]
See comments

File: views/014_job_history_status.sql (hunk 2 above)
Comment (adityathebe) [reply views/014_job_history_status.sql]: It's already like that.  All the columns from `job_history` is fetched. You're looking at the intermediate table I'm using to get the latest history per resource.

```
+--------------------------------------+----------------------------+---------------+-------------+------------------+----------+-----------------+-----------------+--------------------------------------+----------+----------------------------+----------------------------+----------------------------+
| id                                   | name                       | success_count | error_count | details          | hostname | duration_millis | resource_type   | resource_id                          | status   | time_start                 | time_end                   | created_at                 |
|--------------------------------------+----------------------------+---------------+-------------+------------------+----------+-----------------+-----------------+--------------------------------------+----------+----------------------------+----------------------------+----------------------------|
| 018704a4-772d-ae27-c1ad-8cb487249faa | SystemTemplateSync         | 1             | 0           | {"errors": null} | Arch     | 14              | system_template | 01870403-e772-44ee-98fd-0f330d651e4b | FINISHED | 2023-03-21 20:33:20.013208 | 2023-03-21 20:33:20.027398 | 2023-03-21 14:48:20.013447 |
| 018704a4-772d-833f-1c90-359e4908963d | ComponentStatusSummarySync | 2             | 0           | {"errors": null} | Arch     | 19              |                 |                                      | FINISHED | 2023-03-21 20:33:20.012519 | 2023-03-21 20:33:20.032311 | 2023-03-21 14:48:20.012833 |
+--------------------------------------+----------------------------+---------------+-------------+------------------+----------+-----------------+-----------------+--------------------------------------+----------+----------------------------+----------------------------+----------------------------+
```

File: views/014_job_history_status.sql (hunk 3 above)
Comment (adityathebe) [reply views/014_job_history_status.sql]: I see that those are just labels given to the job. Couldn't find one for config scraper

**Review by moshloop — APPROVED** [review]

**Pull request merged**